### DISCUSSION TITLE
Converting 16-bit TIFF to 8-bit TIFF

### DISCUSSION THREAD

**Discussion by CapeHobbit** [ImageMagick/ImageMagick#4489] [Help] [answered]
First off, I am an Imagemagick novice, so sorry if this is a stupid question. I have a series of 16-bit TIFF images that I want to convert to 8-bit to reduce their size. On the face of it, this seems simply achievable using:

`convert output.tif -depth 8 output2.tif
`
...and yet this has no effect. If I dump the metadata from my images, they seem somewhat inconsistent in depth, as below:

Filename: output.tif
  Format: TIFF (Tagged Image File Format)
  Mime type: image/tiff
  Class: DirectClass
  Geometry: 11228x4091+0+0
  Resolution: 1x1
  Print size: 11228x4091
  Units: Undefined
  Colorspace: sRGB
  Type: TrueColorAlpha
  Base type: TrueColor
  Endianness: LSB
  Depth: 8/16-bit
  Channel depth:
    Red: 16-bit
    Green: 16-bit
    Blue: 16-bit
    Alpha: 1-bit
  Channel statistics:
    Pixels: 45933748
    Red:
      min: 0  (0)
      max: 255 (1)
      mean: 87.361563 (0.34259436)
      median: 0 (0)
      standard deviation: 86.769369 (0.34027203)
      kurtosis: -1.3816788
      skewness: 0.4339388
      entropy: 0.74140326
    Green:
      min: 0  (0)
      max: 255 (1)
      mean: 90.967417 (0.35673497)
      median: 0 (0)
      standard deviation: 86.287607 (0.33838277)
      kurtosis: -1.4079172
      skewness: 0.34832836
      entropy: 0.73534329
    Blue:
      min: 0  (0)
      max: 255 (1)
      mean: 90.802547 (0.35608842)
      median: 0 (0)
      standard deviation: 87.56906 (0.34340808)
      kurtosis: -1.4337038
      skewness: 0.35555739
      entropy: 0.74220131
    Alpha:
      min: 0  (0)
      max: 255 (1)
      mean: 163.24204 (0.64016487)
      median: 0 (0)
      standard deviation: 122.38773 (0.47995189)
      kurtosis: -1.6588519
      skewness: -0.58407884
      entropy: 0.94254625
  Image statistics:
    Overall:
      min: 0  (0)
      max: 255 (1)
      mean: 108.09339 (0.42389566)
      median: 0 (0)
      standard deviation: 95.753442 (0.37550369)
      kurtosis: -1.5942575
      skewness: 0.24756989
      entropy: 0.79037353

No matter what I try, I don't seem to be able to change the individual channel depth, and the only way I can reduce the file size and general depth to 8 is to change the type to palette, but that drops the alpha channel. I have tried using -separate and -combine to work each channel individually, -compression none, -auto-level...none help. Am I missing something obvious (e.g. Tiff must be 16-bit with transparency) or subtle (new colourspace)? The actual constituent values in each RGB band are unsigned 8-bit integers (0 - 255), as written by python (using tifffile).

I am running imagemagick on OSx 11.3 (Big Sur), compiled from source with HDRI and 32-bit support as I work with a load of float32 satellite data (though not in this case). Details as follows:

Version: ImageMagick 7.1.0-14 Q32-HDRI x86_64 2021-10-28 https://imagemagick.org
Copyright: (C) 1999-2021 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI 
Delegates (built-in): bzlib djvu fontconfig freetype gvc heic jbig jng jp2 jpeg lcms lzma openexr pangocairo png raw tiff webp x xml zlib
Compiler: gcc (4.2)

**Comment by snibgo** [comment, marked as answer]
> Am I missing something obvious...?

Yes.

> If I dump the metadata from my images,...

That is metadata about the image in memory, after IM has read the file. But do you care about that? I think you care about the bit depth within the file. IM doesn't tell you that. exiftool does.

**Reply by CapeHobbit** [reply]
Thanks snigbo, I naively assumed that Imagemagick was also updating the metadata as it went. Checking with exiftool does give me the expected changes. Thanks again for the help.

**Comment by snibgo** [comment]
Notice the line:
```
Depth: 8/16-bit
```
This means the values of "image->depth" (as metadata read from the file, or modified by the "-depth" setting) is 8, but the current depth as measured by the pixels in memory is 16. (See identify.c function IdentifyImage().)

So this is a hint that the file was 8-bit. But, as a general rule, don't use IM to discover how data is stored in files. Always use exiftool for that.
